### PR TITLE
feat(axios-extension): use new service gen api

### PR DESCRIPTION
**Pull request by docirl**
https://github.com/SAP/open-ux-tools/issues/2063

Use new api endpoints for ui service generator

### CODE REVIEW

File: packages/axios-extension/src/abap/adt-catalog/generators/ui-service-generator.ts [hunk 1]
```diff
@@ -49,42 +66,64 @@ export class UiServiceGenerator extends AdtService {
...
-     * The uri is returned from the generate method.
+     * Generate the service binding.
+     *
+     * @param content - The content of the service binding.
+     * @param transport - The transport.
+     * @returns The object references.
      */
-    public async lockServiceBinding() {
-        await this.post('', '', {
+    public async generate(content: string, transport: string): Promise<any> {
```
Comment (cianmSAP) [inline packages/axios-extension/src/abap/adt-catalog/generators/ui-service-generator.ts:116]: Would the return type be `unknown` rather than any?

**Review by cianmSAP — DISMISSED** [review]
Updates are clear
Changeset ✅
Did not test locallt
One small query

File: packages/axios-extension/src/abap/adt-catalog/generators/ui-service-generator.ts (hunk 1 above)
Comment (docirl) [reply packages/axios-extension/src/abap/adt-catalog/generators/ui-service-generator.ts]: Thanks, updated

**Review by cianmSAP — APPROVED** [review]
Re-approving

File: packages/axios-extension/src/abap/adt-catalog/generators/ui-service-generator.ts [hunk 2]
```diff
@@ -21,6 +21,23 @@ export class UiServiceGenerator extends AdtService {
...
+    public async getSchema(): Promise<any> {
+        const response = await this.get('/schema', {
+            headers: {
+                Accept: 'application/vnd.sap.adt.repository.generator.schema.v1+json'
+            },
+            params: {
+                referencedObject: this.bo.uri
+            }
+        });
+        return JSON.parse(response.data);
```
Comment (devinea) [inline packages/axios-extension/src/abap/adt-catalog/generators/ui-service-generator.ts:38]: try catch?

File: packages/axios-extension/src/abap/adt-catalog/generators/ui-service-generator.ts [hunk 3]
```diff
@@ -49,42 +66,64 @@ export class UiServiceGenerator extends AdtService {
...
+                Accept: 'application/vnd.sap.adt.repository.generator.v1+json, application/vnd.sap.as+xml;charset=UTF-8;dataname=com.sap.adt.StatusMessage'
             },
             params: {
-                _action: `LOCK`,
-                accessMode: 'MODIFY'
+                referencedObject: this.bo.uri,
+                corrNr: transport
             }
         });
+        return JSON.parse(response.data);
```
Comment (devinea) [inline packages/axios-extension/src/abap/adt-catalog/generators/ui-service-generator.ts:127]: try catch?

**Review by devinea — APPROVED** [review]
Changes are clear and cleaner than calling the older API's.
Few small comments on making the code more robust. 
changeset ✅ 
I haven't tested locally.

File: packages/axios-extension/src/abap/adt-catalog/generators/ui-service-generator.ts (hunk 2 above)
Comment (docirl) [reply packages/axios-extension/src/abap/adt-catalog/generators/ui-service-generator.ts:38]: I have a catch for errors from the calling generator, this is also similar to the other methods here for service gen. If it's ok I'll follow that pattern here?

File: packages/axios-extension/src/abap/adt-catalog/generators/ui-service-generator.ts (hunk 3 above)
Comment (docirl) [reply packages/axios-extension/src/abap/adt-catalog/generators/ui-service-generator.ts:127]: I have a catch for errors from the calling generator, this is also similar to the other methods here for service gen. If it's ok I'll follow that pattern here?

File: examples/odata-cli/src/activities.ts [hunk 4]
```diff
@@ -208,45 +207,22 @@ export async function testUiServiceGenerator(
...
-    // Generator service
+    // Get generator service
     const generator = await provider.getUiServiceGenerator(bo);
     const content = await generator.getContent(env.TEST_PACKAGE);
-    logger.debug('content: ' + content);
-    let generatedRefs;
+
+    // Validate package and content
+    logger.info('validate package: ' + JSON.stringify(await generator.validatePackage(env.TEST_PACKAGE)));
+    logger.info('validate content: ' + JSON.stringify(await generator.validateContent(content)));
```
[the remark this answers is not in the document]
Comment (docirl) [reply examples/odata-cli/src/activities.ts:216]: Thanks, this is a test file but I can add a try/catch to here (maybe in a later pr if possible?).

File: packages/axios-extension/src/abap/adt-catalog/generators/ui-service-generator.ts (hunk 2 above)
Comment (devinea) [reply packages/axios-extension/src/abap/adt-catalog/generators/ui-service-generator.ts:38]: OK. Thanks

**Pull request merged**